src/poliastro/plotting/_base.py

```diff
@@ -170,13 +168,27 @@ def _plot_maneuver(
 
         if len(maneuver_phases) == 0:
             # For single-impulse maneuver only draw the impulse marker
-            return ([self._draw_impulse(color, f"Impulse - {label}", final_phase.r)],)
+            impulse_label = f"Impulse 1 - {label}"
+            impulse_lines = ([self._draw_impulse(color, impulse_label, final_phase.r)],)
+            return [(impulse_label, impulse_lines)]
         else:
-            coordinates_list = []
+            # Declare for holding (label, lines) for each impulse and trajectory
+            lines_list = []
 
             # Collect the coordinates for the different maneuver phases
             for ith_impulse, orbit_phase in enumerate(maneuver_phases):
 
+                # Plot the impulse marker and collect its label and lines
+                impulse_label = f"Impulse {ith_impulse + 1} - {label}"
+                impulse_lines = (
+                    [self._draw_impulse(color, impulse_label, orbit_phase.r)],
+                )
+                lines_list.append((impulse_label, impulse_lines))
+
+                # HACK: if no color is provided, get the one randomly generated
+                # for previous impulse lines
+                color = impulse_lines[0][0].get_color() if color is None else color
```

Comment:
This line of code executes every time, reducing the performance...

Reply:
I'm not concerned about the performance of plotting. About your question:

> Should we leave the random color behavior so that when users get the plot they realize providing a color is a good idea?

If `plot_maneuver` plots both markers _and_ lines, I think they should have the same color. If the user specifies a color, then that one should be used. If not, then then color that matplotlib chooses should be used, both for markers and lines. (This is not random, but the next color in the cycle - codenamed "C0", "C1", "C2", ...)

Does this make sense?

Reply:
Sure! Then, previous code accomplishes the suggested task 👍🏽

Reply:
Awesome 💪🏽